File: src/js/services/eventService.js
export async function getEvents() {
  try {
    const response = await fetch("https://maratune.onrender.com/events");
    return response.json();
  } catch (error) {
    console.log(Error, "error");
  }
}

export async function newEvent(
  name,
  description,
  date_event,
  date_time,
  city,
  cups,
  status,
  id_event_organizer,
) {
  try {
    const response = await fetch("https://maratune.onrender.com/events", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        name,
        description,
        date_event,
        date_time,
        city,
        cups: Number(cups),
        status,
        id_event_organizer: Number(id_event_organizer),
      }),
    });

    if (!response.ok) {
      throw new Error("Could not create event");
    }

    const data = await response.json();
    return data;
  } catch (error) {
    console.error(error);
    return null;
  }
}

export async function APIeditEvent(
  event_id,
  name,
  description,
  date_event,
  date_time,
  status,
) {
  try {
    const response = await fetch(
      `https://maratune.onrender.com/events/${event_id}`,
      {
        method: "PATCH",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          name,
          description,
          date_event,
          date_time,
          status,
        }),
      },
    );
    if (!response.ok) {
      throw new Error("Could not edit event");
    }
    const data = await response.json();
    return data;
  } catch (error) {
    console.log(Error, "error");
  }
}

export async function removeEvent(event_id) {
  try {
    const response = await fetch(
      `https://maratune.onrender.com/events/${event_id}`,
      {
        method: "DELETE",
      },
    );

    if (!response.ok) {
      throw new Error("could not delete the event");
    }
    return true;
  } catch (error) {
    console.error(error);
  }
}

export function viewDetails(events) {
  const viewDetails = document.getElementById("viewDetails");
  const btnsDetails = document.querySelectorAll(".details");
  btnsDetails.forEach((btn) => {
    btn.addEventListener("click", () => {
      const idDetails = Number(btn.dataset.id);
      const selectedEvent = events.find((e) => e.id_event === idDetails);

      if (selectedEvent) {
        console.log("entro al if");

        viewDetails.innerHTML = `
                    <div class="modal">
    <div class="modal-content">

        <div class="modal-header">
            <img src="src/assets/logo/only-logo.png" alt="Logo evento">

            <div>
                <h2>${selectedEvent.name}</h2>
                <span class="status ${selectedEvent.status}">${selectedEvent.status}</span>
            </div>
        </div>

        <hr>

        <div class="modal-body">

            <div class="detail-item">
                <span class="label">Description</span>
                <p>${selectedEvent.description}</p>
            </div>

            <div class="detail-item">
                <span class="label">City</span>
                <p>${selectedEvent.city}</p>
            </div>

            <div class="detail-item">
                <span class="label">Date</span>
                <p>${selectedEvent.date_event}</p>
            </div>

            <div class="detail-item">
                <span class="label">Hour</span>
                <p>${selectedEvent.date_time}</p>
            </div>

            <div class="detail-item">
                <span class="label">Cups</span>
                <p>${selectedEvent.cups}</p>
            </div>

        </div>

        <button id="close-modal">
            Close
        </button>

    </div>
</div>
                `;

        document.getElementById("close-modal").addEventListener("click", () => {
          viewDetails.innerHTML = "";
        });
      }
    });
  });
}

File: app/services/event_service.py
from database import get_connection
from fastapi import HTTPException

def create_event(event):

    connection = get_connection()
    cursor = connection.cursor()

    query = """
        INSERT INTO events
        (
            name,
            description,
            date_event,
            date_time,
            city,
            cups,
            status,
            id_event_organizer
        )

        VALUES
        (
            %s,%s,%s,%s,%s,%s,%s,%s
        )

        RETURNING id_event;
    """

    cursor.execute(
        query,
        (
            event.name,
            event.description,
            event.date_event,
            event.date_time,
            event.city,
            event.cups,
            event.status,
            event.id_event_organizer
        )
    )

    new_event = cursor.fetchone()

    connection.commit()

    cursor.close()
    connection.close()

    return {
        "success": True,
        "message": "Event created successfully",
        "id_event": new_event["id_event"],
        "id_event_organizar": event.id_event_organizer
    }

def get_all_events():
    connection = get_connection()
    cursor = connection.cursor()

    query = """
        SELECT *
        FROM events
        ORDER BY date_event;
    """

    cursor.execute(query)

    events = cursor.fetchall()

    cursor.close()
    connection.close()

    return events

def update_event(id_event, event):

    connection = get_connection()
    cursor = connection.cursor()

    cursor.execute(
        "SELECT * FROM events WHERE id_event = %s",
        (id_event,)
    )

    current_event = cursor.fetchone()

    if current_event is None:
        cursor.close()
        connection.close()
        raise HTTPException(status_code=404, detail="Event not found")

    query = """
        UPDATE events
        SET
            name = %s,
            description = %s,
            date_event = %s,
            date_time = %s,
            city = %s,
            cups = %s,
            status = %s
        WHERE id_event = %s
    """

    cursor.execute(
        query,
        (
            event.name if event.name is not None else current_event["name"],
            event.description if event.description is not None else current_event["description"],
            event.date_event if event.date_event is not None else current_event["date_event"],
            event.date_time if event.date_time is not None else current_event["date_time"],
            event.city if event.city is not None else current_event["city"],
            event.cups if event.cups is not None else current_event["cups"],
            event.status if event.status is not None else current_event["status"],
            #event.id_event_organizer if event.id_event_organizer is not None else current_event["id_event_organizer"],
            id_event
        )
    )

    connection.commit()

    cursor.close()
    connection.close()

    return {
        "success": True,
        "message": "Event updated successfully"
    }

def delete_event(id_event):
    connection = get_connection()
    cursor = connection.cursor()

    cursor.execute(
        "SELECT * FROM events WHERE id_event = %s",
        (id_event,)
    )

    event = cursor.fetchone()

    if event is None:
        cursor.close()
        connection.close()
        raise HTTPException(status_code=404, detail="Event not found")

    cursor.execute(
        "DELETE FROM events WHERE id_event = %s",
        (id_event,)
    )

    connection.commit()

    cursor.close()
    connection.close()

    return {
        "success": True,
        "message": "Event deleted successfully"
    }










